### PR TITLE
REF: Vendor ethercatmc databases

### PR DESCRIPTION
**Pull request by klauer**
- [x] Start adding info tags for autosave
- [x] Archiver

### CODE REVIEW

**Comment by klauer** [comment]
If #33 goes in, need to add the archive nodes as well. If not, perhaps generate a separate `.archive` file for the motors.

**Comment by klauer** [comment]
Remove/keep ca09aca based on the archive tool decision

File: app/Db/EthercatMC.template [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+	field(DHLM,"$(DHLM=0)")
+	field(DLLM,"$(DLLM=0)")
+	field(RDBD,"$(RDBD=0)")
+	field(RTRY,"$(RTRY=1)")
+	field(RMOD,"$(RMOD=3)")
+	field(DLY,"$(DLY=0)")
+	field(FOFF,"$(FOFF=0)")
+	field(NTM,"0")
+	field(TWV,"$(TWV=1)")
+    info(autosaveFields_pass0, "VAL DVAL")
```
Comment (ZLLentz) [inline app/Db/EthercatMC.template]: I spy tabs vs spaces

File: app/Db/EthercatMC.template [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+	field(EGU,"$(EGU=mm)")
+	field(DHLM,"$(DHLM=0)")
+	field(DLLM,"$(DLLM=0)")
+	field(RDBD,"$(RDBD=0)")
+	field(RTRY,"$(RTRY=1)")
+	field(RMOD,"$(RMOD=3)")
+	field(DLY,"$(DLY=0)")
+	field(FOFF,"$(FOFF=0)")
+	field(NTM,"0")
+	field(TWV,"$(TWV=1)")
```
Comment (ZLLentz) [inline app/Db/EthercatMC.template]: Should we slip SPDB in here?

File: app/Db/EthercatMC.template [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    info(autosaveFields_pass0, "VAL DVAL")
+    info(autosaveFields, "DIR DESC VELO VBAS VMAX ACCL PREC EGU FOFF OFF")
+    info(archive, "DIR DESC VELO VBAS VMAX ACCL PREC EGU FOFF OFF")
```
Comment (ZLLentz) [inline app/Db/EthercatMC.template]: We also want RBV and VAL archived

File: app/Db/EthercatMC.template [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+	field(DLLM,"$(DLLM=0)")
+	field(RDBD,"$(RDBD=0)")
+	field(RTRY,"$(RTRY=1)")
+	field(RMOD,"$(RMOD=3)")
+	field(DLY,"$(DLY=0)")
+	field(FOFF,"$(FOFF=0)")
+	field(NTM,"0")
+	field(TWV,"$(TWV=1)")
+    info(autosaveFields_pass0, "VAL DVAL")
+    info(autosaveFields, "DIR DESC VELO VBAS VMAX ACCL PREC EGU FOFF OFF")
```
Comment (ZLLentz) [inline app/Db/EthercatMC.template]: Probably also want SPDB if we don't add it above

**Comment by ZLLentz** [comment]
It's not clear to me how many of and which of these files are needed... it's hard to parse and decide on autosave/archive priorities

File: app/Db/EthercatMC.template (hunk 1 above)
Comment (klauer) [reply app/Db/EthercatMC.template]: 😨

File: app/Db/EthercatMC.template (hunk 2 above)
Comment (klauer) [reply app/Db/EthercatMC.template]: Sure - will default it to 0 because of:
https://github.com/epics-modules/motor/blob/855c2540d1b2bd24a9ad2abc72c3f8473d8fa968/motorApp/MotorSrc/motorRecord.cc#L2240

File: app/Db/EthercatMC.template (hunk 3 above)
Comment (klauer) [reply app/Db/EthercatMC.template]: DRBV, RRBV, DVAL, RVAL as well?

**Comment by klauer** [comment]
I'm not entirely sure either, @ZLLentz. It will hopefully become more clear as we use the ESS motor record support more...

**Comment by klauer** [comment]
Oops, looks like the template still needs updating.

File: app/Db/EthercatMC.template (hunk 3 above)
Comment (ZLLentz) [reply app/Db/EthercatMC.template]: I mean why not right?

**Comment by klauer** [comment]
I'll tag + release everything (including a new ads-deploy) so we can give this a try.
